setup.py

```diff
@@ -1,30 +1,168 @@
+#!/usr/bin/env python3
+"""
+python-libzim (aka libzim)
+
+To compile or run this project, you must first get the libzim headers & binary:
+
+ - You can get the headers here and build and install the binary from source:
+   https://github.com/openzim/libzim
+
+ - Or you can download a full prebuilt release (if one exists for your platform):
+   https://download.openzim.org/release/libzim/
+
+   cd /tmp && wget -qO- https://download.openzim.org/release/libzim/libzim_linux-x86_64-6.1.1.tar.gz | tar -xz -C .
+   # to install libzim system-wide, move the headers and dylib into the system paths 
+   mv libzim_linux-x86_64-6.1.1/include/zim /usr/include/zim
+   mv libzim_linux-x86_64-6.1.1/lib/x86_64-linux-gnu/libzim.so.6 /usr/lib/libzim.so
+   # or, set these vars to make the compiler to search specific paths for libzim .h and .so files
+   export LIBZIM_INCLUDE_DIR=/tmp/libzim_linux-x86_64-6.1.1/include
+   export LIBZIM_LIBRARY_DIR=/tmp/libzim_linux-x86_64-6.1.1/lib/x86_64-linux-gnu
+
+To rebuild the cython extension, sdist, and bdist wheels:
+    rm libzim/libzim.cpp
+    rm -rf dist
+    rm -rf build
+    rm -rf libzim.egg-info
+    rm *.so
+    python setup.py build_ext --inplace
+    python setup.py sdist bdist_wheel
+    twine upload --repository-url https://test.pypi.org/legacy/ dist/*
+    pip install --index-url https://test.pypi.org/simple/ libzim
+"""
 import os
-from distutils.core import setup
-from distutils.extension import Extension
+from pathlib import Path
+from setuptools import setup, Extension
+from ctypes.util import find_library
 from Cython.Build import cythonize
 
-# Utility function to read the README file.
-# Used for the long_description.  It's nice, because now 1) we have a top levegit checkout masterl
-# README file and 2) it's easier to type in the README file than to put a raw
-# string in below ...
-def read(fname):
-    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+PACKAGE_NAME = "libzim"
+VERSION = "6.1.1"  # pegged to be the same version as libzim since they are always released together
+LICENSE = "GPLv3"
+DESCRIPTION = "The official Python API for creating and interacting with ZIM files using libzim"
+AUTHOR = "Monadical Inc."
+AUTHOR_EMAIL = "jdc@monadical.com"
+GITHUB_URL = "https://github.com/openzim/python-libzim"
+
+BASE_DIR = Path(__file__).parent
+LIBZIM_CYTHON_DIR = 'libzim'
+
+LIBZIM_INCLUDE_DIR = os.getenv('LIBZIM_INCLUDE_DIR')
+LIBZIM_LIBRARY_DIR = os.getenv('LIBZIM_LIBRARY_DIR')
+
+INCLUDE_DIRS = [str(BASE_DIR / LIBZIM_CYTHON_DIR), '/usr/include']
+LIBRARY_DIRS = ['/usr/lib']
```

Comment:
You should not assume the default include and library dir.
Depending on the system, the default lib dir can be `/usr/lib`, `/usr/lib64`, `/usr/lib/linux-x86_64`, ...

Reply:
changed it to use a relative `./lib` and `./include`